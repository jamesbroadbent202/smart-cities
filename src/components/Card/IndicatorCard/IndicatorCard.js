import React from 'react';
import PropTypes from 'prop-types';
import numeral from 'numeral';
import Card from '../Card';
import Icon from '../../Icon/Icon';
import {
  CARD_SIZES,
  INDICATORS,
  INDICATOR_CARD_TYPES,
} from '../../../constants';
import stripPrefixAndSuffix from '../../../helpers/stripPrefixAndSuffix';
import getColorVariant from '../../../helpers/getColorVariant';
import style from './IndicatorCard.scss';

const classnames = require('classnames/bind').bind(style);

const IndicatorCard = (props) => {
  const indicator = typeof props.indicator === 'string'
    ? INDICATORS[props.indicator]
    : props.indicator;

  const className = classnames(
    style.wrapper,
    `wrapper__${props.size}`,
    props.className,
  );

  // Most indicators define a format. If supplied, this format is applied to the number here
  const formattedNumber = indicator.format
    ? numeral(props.value).format(indicator.format)
    : props.value;

  // The prefix and suffix are rendered in different spans, so we split them out here
  const [formatPrefix, value, formatSuffix] = stripPrefixAndSuffix(formattedNumber);

  // In addition to the format prefix/suffix (e.g. "%")
  // a card specific prefix/suffix can be added (e.g. "per year")
  const displayPrefix = `${indicator.cardPrefix || ''}${formatPrefix}`;
  const displaySuffix = `${formatSuffix}${indicator.cardSuffix || ''}`;

  return (
    <Card
      {...props}
      type={INDICATOR_CARD_TYPES.INDICATOR}
      className={className}
    >
      <p className={style.header}>{indicator.name}</p>

      {!indicator.contextual && (
        <Icon
          className={style.indicatorTypeMark}
          color={getColorVariant(props.colorName, '900')}
          icon={props.isCategoryPage ? 'indicatorTypeMarkSolid' : 'indicatorTypeMarkBorder'}
          size={14}
        />
      )}

      <div className={style.indicatorWrapper}>
        {!!displayPrefix && (
          <span className={classnames(style.prefix, style.symbol)}>{displayPrefix}</span>
        )}

        <span className={style.number}>{value}</span>

        {!!displaySuffix && (
          <span className={classnames(style.suffix, style.symbol)}>{displaySuffix}</span>
        )}
      </div>
    </Card>
  );
};

IndicatorCard.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string.isRequired, // a hex code
  colorName: PropTypes.string.isRequired, // a color base name, e.g. 'jobs'
  indicator: PropTypes.oneOfType([ // accepts an indicator object or a string
    PropTypes.shape({
      cardPrefix: PropTypes.string,
      cardSuffix: PropTypes.string,
      format: PropTypes.string,
      name: PropTypes.string.isRequired,
      contextual: PropTypes.bool.isRequired,
    }),
    PropTypes.string,
  ]).isRequired,
  isCategoryPage: PropTypes.bool,
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]).isRequired,
  size: PropTypes.oneOf(Object.values(CARD_SIZES)),
};

IndicatorCard.defaultProps = {
  size: CARD_SIZES.SMALL,
};

export default IndicatorCard;
