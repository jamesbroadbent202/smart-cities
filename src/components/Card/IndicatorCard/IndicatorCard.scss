@import '~style/variables';

.wrapper {
  display: flex;
  align-items: stretch;
  position: relative; // the P is absolute, relative to this

  &.wrapper__large {
    flex-direction: column;
  }
}

.header {
  flex: 1;

  .wrapper__small & {
    @include type-indicator-card-header--small;
  }

  .wrapper__large & {
    @include type-indicator-card-header--large;
  }
}

.indicatorTypeMark {
  position: absolute;
  top: 7px;
  right: 7px;
}

.indicatorWrapper {
  margin-left: auto; // push right
  align-self: flex-end; // push down
  line-height: 1;
}

.symbol {
  color: $color-grey--500;

  .wrapper__small & {
    @include type-indicator-card-symbol--small;
  }

  .wrapper__large & {
    @include type-indicator-card-symbol--large;
  }
}

.prefix {
  margin-right: 5px;
}

.number {
  .wrapper__small & {
    @include type-indicator-card-number--small;
  }

  .wrapper__large & {
    @include type-indicator-card-number--large;
  }
}

.suffix {
  margin-left: 5px;
}
