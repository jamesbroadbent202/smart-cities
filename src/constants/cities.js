/* eslint-disable */

// This data is used when converting the source data to a format for use in the front end
// The key for each city must match exactly what appears in the source data
// An error will be thrown during conversion if they don't match
export default {
  'Albury - Wodonga': {
    id: 'albury-wodonga',
    shortName: 'Albury/Wodonga',
    description: 'TODO enter in a description for Albury - Wodonga',
  },
  'Australian Capital Territory': {
    id: 'act',
    shortName: 'ACT',
    description: 'TODO enter in a description for Australian Capital Territory',
  },
  'Ballarat': {
    id: 'ballarat',
    shortName: 'Ballarat',
    description: 'TODO enter in a description for Ballarat',
  },
  'Bendigo': {
    id: 'bendigo',
    shortName: 'Bendigo',
    description: 'TODO enter in a description for Bendigo',
  },
  'Cairns': {
    id: 'cairns',
    shortName: 'Cairns',
    description: 'TODO enter in a description for Cairns',
  },
  'Geelong': {
    id: 'geelong',
    shortName: 'Geelong',
    description: 'TODO enter in a description for Geelong',
  },
  'Gold Coast - Tweed Heads': {
    id: 'gold-coast-tweed-heads',
    shortName: 'Gold Coast/Tweed Heads',
    description: 'TODO enter in a description for Gold Coast - Tweed Heads',
  },
  'Greater Adelaide': {
    id: 'greater-adelaide',
    shortName: 'Adelaide',
    description: 'TODO enter in a description for Greater Adelaide',
  },
  'Greater Brisbane': {
    id: 'greater-brisbane',
    shortName: 'Brisbane',
    description: 'TODO enter in a description for Greater Brisbane',
  },
  'Greater Darwin': {
    id: 'greater-darwin',
    shortName: 'Darwin',
    description: 'TODO enter in a description for Greater Darwin',
  },
  'Greater Hobart': {
    id: 'greater-hobart',
    shortName: 'Hobart',
    description: 'TODO enter in a description for Greater Hobart',
  },
  'Greater Melbourne': {
    id: 'greater-melbourne',
    shortName: 'Melbourne',
    description: 'TODO enter in a description for Greater Melbourne',
  },
  'Greater Perth': {
    id: 'greater-perth',
    shortName: 'Perth',
    description: 'Perth is Western Australiaâ€™s state capital city. Perth has an estimated population of over 2.04 million people, which account for 8.58% of the national population. Perth is Australia 4th largest city in population size. Perth is part of the South West Land Division of Western Australia, with the majority of the metropolitan area of Perth located on the Swan Coastal Plain.',
  },
  'Greater Sydney': {
    id: 'greater-sydney',
    shortName: 'Sydney',
    description: 'TODO enter in a description for Greater Sydney',
  },
  'Launceston': {
    id: 'launceston',
    shortName: 'Launceston',
    description: 'TODO enter in a description for Launceston',
  },
  'Mackay': {
    id: 'mackay',
    shortName: 'Mackay',
    description: 'TODO enter in a description for Mackay',
  },
  'Newcastle - Maitland': {
    id: 'newcastle-maitland',
    shortName: 'Newcastle/Maitland',
    description: 'TODO enter in a description for Newcastle - Maitland',
  },
  'Sunshine Coast': {
    id: 'sunshine-coast',
    shortName: 'Sunshine Coast',
    description: 'TODO enter in a description for Sunshine Coast',
  },
  'Toowoomba': {
    id: 'toowoomba',
    shortName: 'Toowoomba',
    description: 'TODO enter in a description for Toowoomba',
  },
  'Townsville': {
    id: 'townsville',
    shortName: 'Townsville',
    description: 'TODO enter in a description for Townsville',
  },
  'Western Sydney': {
    id: 'western-sydney',
    shortName: 'Western Sydney',
    description: 'TODO enter in a description for Western Sydney',
  },
  'Wollongong': {
    id: 'wollongong',
    shortName: 'Wollongong',
    description: 'TODO enter in a description for Wollongong',
  },
};
