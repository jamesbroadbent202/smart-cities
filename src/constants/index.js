export * from './misc';
export { default as INDICATORS } from './indicators';
export { default as CATEGORIES } from './categories';
