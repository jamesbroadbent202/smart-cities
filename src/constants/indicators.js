import {
  DATA_TYPES,
  AGGREGATION_METHODS,
} from './misc';

// Different length text is used like so:
// name: shows in an indicator card and as the
// title of a chart (or in the legend for stacked charts)
// shortDescription: shows on the axis of a chart
// longDescription: shows in the tooltip for a chart
export default {
  participationRate: {
    name: 'Participation rate',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Participation rate, % of population aged over 15 years',
  },
  gdp: {
    name: 'Economic output per capita',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '$0,0.0a',
    source: 'Gross domestic product, AUD per capita per year',
  },
  educationalAttainmentYr12: {
    name: 'Educational attainment - year 12',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Education attainment - school completion rate, Year 12 or equivalent, % of population',
  },
  population: {
    name: 'The population',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.SUM,
    format: '0[.]0a',
    source: 'Population, All persons, People',
  },
  growthRate: {
    name: 'The growthRate',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    cardSuffix: '/year',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Population growth rate, All persons, % per year',
  },
  dispersion: {
    name: 'Local government dispersion',
    shortDescription: 'Local government dispersion, Share of LGAs in bottom 10% by population',
    longDescription: 'Local government dispersion, Share of LGAs in bottom 10% by population',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Local government dispersion, Share of LGAs in bottom 10% by population',
  },
  roadCongestionCost: {
    name: 'Annual cost of road congestion',
    shortDescription: 'Annual cost of road congestion, AUD Billions per year, indexed to 2010',
    longDescription: 'Annual cost of road congestion, AUD Billions per year, indexed to 2010',
    cardSuffix: 'AUD',
    chartPrefix: '$',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.SUM,
    format: '$0,0[.]0a',
    source: 'Annual cost of road congestion , AUD Billions per year, indexed to 2010',
  },
  dwellingHouse: {
    name: 'House',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Separate house, Percentage of Dwellings',
  },
  dwellingSemi: {
    name: 'Semi-House',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Semi-detached, Percentage of Dwellings',
  },
  dwellingApartment: {
    name: 'Apartment',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Flat or Apartment, Percentage of Dwellings',
  },
  dwellingOther: {
    name: 'Other',
    shortDescription: 'TODO',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Dwelling type ratio, Other, Percentage of Dwellings',
  },
  jobsWalk: {
    name: 'Proportion of jobs accessible within 30 minutes by Walking',
    shortDescription: 'Proportion of jobs accessible within 30 minutes, Walking, % of all jobs in city',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Proportion of jobs accessible within 30 minutes, Walking, % of all jobs in city',
  },
  jobsBike: {
    name: 'Jobs accessible within 30 minutes by bike',
    shortDescription: 'Jobs accessible within 30 minutes by bike',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Proportion of jobs accessible within 30 minutes, By Bicycle, % of all jobs in city',
  },
  jobsCar: {
    name: 'Proportion of jobs accessible within 30 minutes by Car',
    shortDescription: 'Proportion of jobs accessible within 30 minutes, By Car, % of all jobs in city',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Proportion of jobs accessible within 30 minutes, By Car, % of all jobs in city',
  },
  activeJourneys: {
    name: 'Active transport use',
    shortDescription: 'Active transport use, % of journeys to work',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Active transport use, % of journeys to work',
  },
  publicJourneys: {
    name: 'Public transport use',
    shortDescription: 'Public transport use, % of journeys to work',
    longDescription: 'TODO',
    dataType: DATA_TYPES.NUMBER,
    aggregationMethod: AGGREGATION_METHODS.MEDIAN,
    format: '0[.]0%',
    source: 'Public transport use, % of journeys to work',
  },
};
