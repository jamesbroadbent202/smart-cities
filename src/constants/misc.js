export const DATA_URL = '/data/smartCitiesData.json';

export const NO_CATEGORY = 'overview';
export const NO_CITY = 'all-cities';

export const CARD_SIZES = {
  SMALL: 'small',
  LARGE: 'large',
};

export const INDICATOR_CARD_TYPES = {
  CATEGORY: 'CATEGORY',
  INDICATOR: 'INDICATOR',
};

// note, the values are lowercase because they're used in CSS class names
export const COLOR_NAMES = {
  CONTEXT: 'context',
  JOBS: 'jobs',
  HOUSING: 'housing',
  INFRASTRUCTURE: 'infrastructure',
  LIVEABILITY: 'liveability',
  INNOVATION: 'innovation',
  PLANNING: 'planning',
};

export const AGGREGATION_METHODS = {
  MEDIAN: 'MEDIAN',
  SUM: 'SUM',
};

export const DATA_TYPES = {
  NUMBER: 'NUMBER',
  TEXT: 'TEXT',
};

export const LINKS = {
  ORG: 'https://dta.gov.au/',
  CONTACT_US: 'mailto:performance-dashboard@digital.gov.au',
};

export const STRINGS = {
  OVERVIEW_TITLE: 'Smart cities',
  OVERVIEW_DESCRIPTION: 'The National Cities Performance Framework provides a snapshot of the productivity, liveability and progress of Australia\'s major cities and tracks changes in their performance over time. The Performance Framework is the first of its kind for Australia, bringing together critical cities information in an easily accessible online format, in the one location.',
};
