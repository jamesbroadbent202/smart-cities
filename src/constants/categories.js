import { COLOR_NAMES } from './misc';

export default [
  {
    id: 'jobs',
    colorName: COLOR_NAMES.JOBS,
    name: 'Jobs & Skills',
    description: 'Jobs and Skills encompasses all key elements of employment and training in our cities, including the performance of the employment market and the skill level of the workforce.',
    subCategories: [
      {
        name: 'Employment',
        tint: '040',
        shade: '600',
        indicatorIds: [
          'participationRate',
        ],
      },
      {
        name: 'Education',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'educationalAttainmentYr12',
        ],
      },
    ],
    heroIndicatorId: 'population',
  },
  {
    id: 'housing',
    colorName: COLOR_NAMES.HOUSING,
    name: 'Housing',
    description: 'Housing encompasses three broad dimensions: the affordability of housing in our cities; the supply and diversity of new housing stock; and where housing is located, including how accessible it is to jobs and services.',
    subCategories: [
      {
        name: 'Buying houses',
        tint: '040',
        shade: '600',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
      {
        name: 'Living in houses',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
    ],
    heroIndicatorId: 'populationGrowth',
  },
  {
    id: 'infrastructure',
    colorName: COLOR_NAMES.INFRASTRUCTURE,
    name: 'Infrastructure & Other Things',
    navName: 'Infrastructure',
    description: 'Infrastructure and Investment encompasses all key dimensions of the cityâ€™s investment environment, with a particular focus on the quality, efficiency and effectiveness of infrastructure.',
    subCategories: [
      {
        name: 'Getting to work',
        tint: '040',
        shade: '600',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
      {
        name: 'Getting around',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'activeJourneys',
          'publicJourneys',
        ],
      },
      {
        name: 'Infrastructure',
        tint: '080',
        shade: '800',
        indicatorIds: [],
      },
    ],
    heroIndicatorId: 'population',
  },
  {
    id: 'liveability',
    colorName: COLOR_NAMES.LIVEABILITY,
    name: 'Liveability',
    description: 'Liveability and Sustainability encompasses three broad dimensions: the health and wellbeing of residents; the attractiveness and amenity of the city; and the state of the environment and the local response to climate change.',
    subCategories: [
      {
        name: 'Environment',
        tint: '040',
        shade: '600',
        indicatorIds: [],
      },
      {
        name: 'Public safety',
        tint: '060',
        shade: '700',
        indicatorIds: [],
      },
      {
        name: 'Life and death',
        tint: '080',
        shade: '800',
        indicatorIds: [],
      },
      {
        name: 'Socioeconomic disadvantage',
        tint: '100',
        shade: '900',
        indicatorIds: [],
      },
    ],
    heroIndicatorId: 'dispersion',
  },
  {
    id: 'innovation',
    colorName: COLOR_NAMES.INNOVATION,
    name: 'Innovation',
    description: 'Innovation and Digital Opportunities encompasses three broad dimensions: city productivity; innovation and entrepreneurship; and access to public data',
    subCategories: [
      {
        name: 'Data',
        tint: '040',
        shade: '600',
        indicatorIds: [],
      },
      {
        name: 'Working in digital',
        tint: '060',
        shade: '700',
        indicatorIds: [],
      },
    ],
    heroIndicatorId: 'population',
  },
  {
    id: 'planning',
    colorName: COLOR_NAMES.PLANNING,
    name: 'Planning',
    description: 'Governance, Planning and Regulation encompasses land use planning in cities and its administration, as well as how effectively local governance and regulation support economic, social and environmental outcomes.',
    subCategories: [
      {
        name: 'Development',
        tint: '040',
        shade: '600',
        indicatorIds: [],
      },
      {
        name: 'Population',
        tint: '060',
        shade: '700',
        indicatorIds: [
          'jobAccessibility',
        ],
      },
    ],
    heroIndicatorId: 'population',
  },
];
