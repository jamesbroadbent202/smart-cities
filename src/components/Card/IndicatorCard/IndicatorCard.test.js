import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import IndicatorCard from './IndicatorCard';
import { CARD_SIZES } from '../../../constants';

jest.mock('../../Icon/Icon', () => 'Icon');

const defaultProps = {
  value: 77,
  colorName: 'jobs',
  indicator: {
    name: 'The population growth',
    cardPrefix: 'The',
    cardSuffix: 'per annum',
    format: '0[.]0%',
    contextual: false,
  },
  color: '#fff',
};

it('should match Snapshot', () => {
  const component =  renderer.create(<IndicatorCard {...defaultProps} />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('should apply the correct size class', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      size={CARD_SIZES.LARGE}
    />
  );

  expect(component.hasClass('largeCard')).toBe(true);
});

it('should handle an indicator being passed as a string', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      indicator="population"
    />
  );

  expect(component.find('.header').text()).toBe('Population');
});

it('should render no prefix or suffix when none are available', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      indicator={{
        name: 'The number of cats',
        cardPrefix: '',
        cardSuffix: '',
        format: '0[.]0',
        contextual: false,
      }}
    />
  );

  expect(component.find('.prefix').length).toBe(0);
  expect(component.find('.suffix').length).toBe(0);
});

it('should render the format prefix', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      indicator={{
        name: 'The cost of things',
        cardPrefix: '',
        cardSuffix: '',
        format: '$0[.]0',
        contextual: false,
      }}
    />
  );

  expect(component.find('.prefix').text()).toBe('$');
});

it('should render the card prefix', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      indicator={{
        name: 'The cost of things',
        cardPrefix: 'AUD ',
        cardSuffix: '',
        format: '$0[.]0',
        contextual: false,
      }}
    />
  );

  expect(component.find('.prefix').text()).toBe('AUD $');
});

it('should render the format suffix', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      indicator={{
        name: 'Population growth',
        cardPrefix: '',
        cardSuffix: '',
        format: '0[.]0%',
        contextual: false,
      }}
    />
  );

  expect(component.find('.suffix').text()).toBe('%');
});

it('should render the card suffix', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      indicator={{
        name: 'Population growth',
        cardPrefix: '',
        cardSuffix: '/year',
        format: '0[.]0%',
        contextual: false,
      }}
    />
  );

  expect(component.find('.suffix').text()).toBe('%/year');
});

it('should render an Icon if the card is not on the context page', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      isContextPage={false}
    />
  );

  expect(component.find('Icon').length).toBe(1);
});

it('should not render an Icon if the card is on the context page', () => {
  const component =  shallow(
    <IndicatorCard
      {...defaultProps}
      isContextPage
    />
  );

  expect(component.find('Icon').length).toBe(0);
});
