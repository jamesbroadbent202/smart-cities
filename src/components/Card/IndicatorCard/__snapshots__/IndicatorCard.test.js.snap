// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<div
  className="card card__small wrapper wrapper__small"
  style={
    Object {
      "borderBottomColor": "#fff",
    }
  }
>
  <p
    className="header"
  >
    The population growth
  </p>
  <Icon
    className="indicatorTypeMark"
    color="JOBS_900"
    icon="indicatorTypeMarkBorder"
    size={14}
  />
  <div
    className="indicatorWrapper"
  >
    <span
      className="prefix symbol"
    >
      The
    </span>
    <span
      className="number"
    >
      7700
    </span>
    <span
      className="suffix symbol"
    >
      %per annum
    </span>
  </div>
</div>
`;
